core/spec/models/spree/payment_method/store_credit_spec.rb

```diff
@@ -0,0 +1,282 @@
+      amount: amount,
+      authorization_code: auth_code)
+
+      response = subject.purchase(amount * 100.0, store_credit, gateway_options)
+      expect(response.success?).to be false
+      expect(response.message).to include Spree.t('store_credit_payment_method.unable_to_find')
+    end
+
+    it "captures a purchase if it can find a pending credit for the correct amount" do
+      amount = 100.0
+      store_credit = create(:store_credit)
+      auth_code = store_credit.generate_authorization_code
+      store_credit.store_credit_events.create!(action: Spree::StoreCredit::ELIGIBLE_ACTION,
+      amount: amount,
+      authorization_code: auth_code)
+
+      response = subject.purchase(amount * 100.0, store_credit, gateway_options)
+      expect(response.success?).to be true
+      expect(response.message).to include Spree.t('store_credit_payment_method.successful_action', action: Spree::StoreCredit::CAPTURE_ACTION)
+    end
+  end
+
+  context "#credit" do
+    subject do
+      Spree::PaymentMethod::StoreCredit.new.credit(credit_amount, auth_code, gateway_options)
+    end
+
+    let(:credit_amount) { 100.0 }
+    let(:auth_code) { auth_event.authorization_code }
+    let(:gateway_options) { super().merge(originator: originator) }
+    let(:auth_event) { create(:store_credit_auth_event) }
+    let(:originator) { nil }
+
+    context 'with an invalid auth code' do
+      let(:auth_code) { 1 }
+
+      it "declines an unknown store credit" do
+        expect(subject.success?).to be false
+        expect(subject.message).to include Spree.t('store_credit_payment_method.unable_to_find')
+      end
+    end
+
+    context "when the store credit isn't credited successfully" do
+      before { allow(Spree::StoreCredit).to receive_messages(credit: false) }
+
+      it "returns an error response" do
+        expect(subject.success?).to be false
+      end
+    end
+
+    context 'with a valid credit request' do
+      before { allow_any_instance_of(Spree::StoreCredit).to receive_messages(credit: true) }
+
+      it "credits a valid store credit credit request" do
+        expect(subject.success?).to be true
+        expect(subject.message).to include Spree.t('store_credit_payment_method.successful_action', action: Spree::StoreCredit::CREDIT_ACTION)
+      end
+    end
+
+    context 'with an originator' do
+      let(:originator) { double('originator') }
+
+      it 'passes the originator' do
+        allow_any_instance_of(Spree::StoreCredit).to receive(:credit).with(anything, anything, anything, action_originator: originator)
+        subject
+      end
+    end
+  end
+
+  context "#cancel" do
+    subject do
+      Spree::PaymentMethod::StoreCredit.new.cancel(auth_code)
+    end
+
+    let(:store_credit) { create(:store_credit, amount_used: captured_amount) }
+    let(:auth_code)    { "1-SC-20141111111111" }
+    let(:captured_amount) { 10.0 }
+
+    let!(:capture_event) { create(:store_credit_auth_event,
```

Comment:
Avoid using `{...}` for multi-line blocks.<br>Block body expression is on the same line as the block start.